### ISSUE TITLE
`Start dev` starts debug mode. 

### ISSUE THREAD

**Issue by kadel** [redhat-developer/vscode-openshift-tools#2752]
When I execute "Start Dev" action it executes `odo dev --debug` but that is not what I want.
I need to execute the "run" command from the devfile, not "debug".

There should be a separate action for starting `odo dev --debug`. Something like "Start Dev with Debug"

**Comment by jeffmaury** [comment]
This is by design and it simplifies the workflow. IntelliJ OpenShift Toolkit and JBoss Tools does the same

**Comment by dgolovin** [comment]
@kadel We can do that, but as @jeffmaury said it simplifies the workflow, because development assumes you might need to debug your app.

Would you explain why you need 'run' not 'debug' to be executed?

**Comment by kadel** [comment]
It is more about what I, as a user, would expect. In IDEs the debug is not started automatically either. You always have a separate action, "Run" and "Debug". 

The main issue is that the menu item is called "Start Dev", so ending up with the debug surprised me. I guess just changing the label to "Start Debug" would fix that. But I still think that there should be an option to give users a choice of what should be done.

I don't want always start my application in debug mode. Not everyone uses debuggers, and also, in some cases debugging modes can have side-effects.  In the current situation, I'm not even aware that I'm running my app with debug enabled. 


I haven't tried that, but I assume that if the Devfile doesn't have `debug` command, then it falls back to `run`. This could be also an issue as running the same action results in different outcomes depending on the defile. 
As a user, I could be using devfile with `debug`, and get used to the fact that debug is always executed. Then I switch projects and this one doesn't have `debug` in devfile. I run it in the same way, but now I'm left wondering what is happening and where is my debugger. 

If there are two separate items `Start Dev` and `Start Debug` the debug option can be simply hidden when there is no `debug` command in the devfile. This way it would be clear to the user that in this project the debugger is not defined.

**Comment by jeffmaury** [comment]
I think you missed one point: we are not starting the debugger when we run Start dev but we are only running odo in debug mode. Then it's up to the user to launch the debugger through the Debug menu item.
I think you are viewing this from an odo expert POV and not from a lambda user POV.
Please note that if the delay running odo in debug mode that will cause delays to the user when debug is to be started because it will perform the following actions

- stop odo dev currently running (if any) -> long additional time
- start odo dev --debug
- ensure debug port is available
- start debugger

**Comment by kadel** [comment]
I'm talking about clearly communicating to the user when the application is run in debug mode and when not. The user should be in control when the debugging is enabled and when not. 


> * stop odo dev currently running (if any) -> long additional time
> * start odo dev --debug
> * ensure debug port is available
> * start debugger

This is what I, as the IDE user, would expect because this is what I'm used to from my IDE.
If I just "Run" my application, then I expect that I will have to stop it and start again when I need to enable debug. If I start it in "Debug" from the beginning, than I don't have to do that. But I should decide what I want to do. In IntelliJ, you also have two options, either "Run" or "Debug".


In every IDE I have the option to run applications normally or with debugging enabled. We should give users the option as well. Starting the application with debug enabled, especially if the menu item doesn't mention anything about debugging is against what I, as the user expect.

**Comment by jeffmaury** [comment]
Yes but the user is in control when he wants to debug or not and I think this is the main point

**Comment by kadel** [comment]
> Yes but the user is in control when he wants to debug or not, and I think this is the main point

No, he is not. The application is always started with debugging enabled. The user is only in control when he/she starts the debugging session. Important is to have control over how the application is started (with or without debug)

**Comment by kadel** [comment]
> That's an odo view for me. 

That has nothing to do with odo. This is how every IDE works. 


> That's an odo view for me. Look at how at Quarkus application is started, debugger is enabled by default, that was a good decision by odo team that's the same pattern that we implemented ie start app then connect the debbugger if requested

Not sure I understand what you mean :-/ To enable debug mode in quarkus you have to pass explicitly `-Ddebug`. Which is not there in the default `run` command in quarkus devfile. `odo dev` uses `run` command. So by default, it is not started with the debugger enabled. 


But we are getting off-topic. odo aside. My main point is that every IDE user is used to the "run" and "debug" options in their IDE. **The user is the one who decides whether to enable debug mode in-app or not**

IntelliJ IDEA

<img width="428" alt="Screenshot 2023-01-27 at 10 03 30" src="https://user-images.githubusercontent.com/57206/215048206-e7535ad9-cb12-43c3-a601-207968c532bb.png">

VSCode

<img width="272" alt="Screenshot 2023-01-27 at 10 05 33" src="https://user-images.githubusercontent.com/57206/215048625-f10b92e2-4bff-42eb-9dd0-61920189c337.png">

VSCode even makes it look like "Start Debugging" is the preferred way to run. And "Run Without Debugging" is an alternative.

Our plugins should give users the same option.
"Start dev mode"/"Start dev mode with debugging"

Or we can invert it and have "Start dev mode"/"Start dev mode without debugging". This will nudge users toward enabling debug mode. Important think is that the user should have and option to enable or not to enable debugging.

**Comment by kadel** [comment]
Now I understand what you meant with quarkus running just `mvn quarkus:dev` enables debug mode. But this is specific to Quarkus. My point still stands. Users should have the option to run applications without enabling debug mode. If the default is with debug enabled it is ok, but there needs to be an option to run it without debug as well.

**Comment by dgolovin** [comment]
@kadel  I see your point and I agree we should do the same what the IDEs do. I was not paying attention what that F5 in vs code is Start Debugging and there is ^ F5 which is just Start.

I also like the idea to get rid of `Dev` in context menus and use just `Run` and `Debug`, 'Run on Podman` and `Debug on Podman`. `Debug` will only show up in case it is declared as supported in `odo component describe`.  We would also have to rename 'Debug' command we have to connect debugger to running component to something else like `Connect Debugger`.

We need better `--platform` support in future because adding context menu 'Run on PlatformN` for N supported platform is obviously not sustainable.

**Comment by kadel** [comment]
> We need better `--platform` support in future because adding context menu 'Run on PlatformN` for N supported platform is obviously not sustainable.

I wouldn't worry about that for now. I don't think we will have more platforms in the foreseeable future.
